### PR TITLE
Update getRootTreeNodeAsync return type

### PR DESCRIPTION
**Pull request by alexgomezlf**
- getRootTreeNodeAsync should throw if the rootTreeNode doesn't exist, not return undefined

### CODE REVIEW

File: projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts [hunk 1]
```diff
@@ -15,28 +15,28 @@ export interface LfTreeNodePage {
 export interface LfTreeNodeService {
```
Comment (paolo-arge-lf) [inline projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts:18]: /**
A tree data structure provider that can be used as a data source for UI components, such as the lf-repository-browser component,  to navigate a tree structure.
*/

File: projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
+  page: LfTreeNode[];
+  nextPage: string | undefined;
 }
 
+/**
+ * A tree data structure provider that can be used as a data source for UI components, such as the lf-repository-browser component, to navigate a tree structure.
+ */
 export interface LfTreeNodeService {
+  /**
+   * Gets LfTreeNode data for the lf-repository-browser to display
```
Comment (paolo-arge-lf) [inline projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts]: Returns a page of children for the parent folder. The implementation may provide expose ordering and filtering functionality.

File: projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
+   * @param folder: string that represents the ID of the folder to get data from
+   * @param nextPage: string representing the next page requested
```
Comment (paolo-arge-lf) [inline projects/ui-components/lf-repository-browser/ILfTreeNodeService.ts]: @param nextPage: string representing the next page requested. If undefined, the first page is returned.

**Review by paolo-arge-lf — DISMISSED** [review]

File: projects/lf-documentation/src/app/lf-repository-browser-documentation/demo-repo-service.ts [hunk 4]
```diff
@@ -304,14 +304,18 @@ export class DemoRepoService implements LfTreeNodeService {
...
-  getTreeNodeByIdAsync(id: string): Promise<LfTreeNode | undefined> {
-    return Promise.resolve(this._entries[id]);
+  async getParentTreeNodeAsync(entry: LfTreeNode): Promise<LfTreeNode | undefined> {
+    if(entry.path === '//') {
+      return undefined;
+    }
+    const path = entry.path.split('//');
+    path.pop(); // remove self
+    const parent = path.pop();
+    const parentId = parent ? this._entries[parent] : this._rootEntry;
```
Comment (alexgomezlf) [inline projects/lf-documentation/src/app/lf-repository-browser-documentation/demo-repo-service.ts]: change name

File: projects/lf-documentation/src/app/lf-repository-browser-documentation/lf-repository-browser-documentation.component.html [hunk 5]
```diff
@@ -113,7 +113,7 @@
   </tr>
   <tr>
     <td>initAsync</td>
-    <td>(providers: LfTreeNodeService, selectedEntryId?: string) => Promise&lt;void&gt;</td>
+    <td>(treeNodeService: LfTreeNodeService, selectedEntry?: LfTreeNode) => Promise&lt;void&gt;</td>
     <td>
       initializes the lf-repository-browser with a LfRepositoryProvider You <b>must</b> call this method. Each time this is called, the
```
Comment (alexgomezlf) [inline projects/lf-documentation/src/app/lf-repository-browser-documentation/lf-repository-browser-documentation.component.html:118]: update description

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 6]
```diff
@@ -340,14 +344,15 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
       this.hasError = true;
       throw new Error('Repository Browser cannot be initialized without a data service.');
     }
-    const rootEntry: LfTreeNode | undefined = await this.treeNodeService.getRootTreeNodeAsync();
-    if (rootEntry == null) {
-      console.error(`Repository browser does not contain a root entry`);
+    try {
+      const rootEntry: LfTreeNode = await this.treeNodeService.getRootTreeNodeAsync();
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: remove initializeWithRootOpne

File: projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts [hunk 7]
```diff
@@ -279,27 +282,28 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
...
+        this.hasError = true;
+        return;
       }
-      currentEntry = currentIdOrEntry;
     }
     // If the entry passed in is not a container we will get the parent of this by default.
     if (currentEntry && !currentEntry.isContainer) {
       currentEntry = await this.treeNodeService.getParentTreeNodeAsync(currentEntry);
     }
+    if (currentEntry?.id == null) {
```
Comment (alexgomezlf) [inline projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts]: !currentEntry

**Review by Ke-Jin-Laserfiche — APPROVED** [review]

**Pull request merged**